src/markup.c

```diff
@@ -242,6 +243,67 @@ char *markup_strip(char *str)
         return str;
 }
 
+/**
+ * Determine if an & character pointed to by \p str is a markup & entity or
+ * part of the text
+ *
+ * @return true if it's an entity otherwise false
+ */
+bool markup_is_entity(char *str)
+{
+        assert(str);
+        assert(*str == '&');
+
+        char *end = strchr(str, ';');
+        if (!end)
```

Comment:
What about the input of `&;`?


```suggestion
        if (!end || end <= str+1)
```

Reply:
This is handled already by the named entity case (`g_str_has_prefix` fails for all supported tags with `&;` as the input).